File: dev-docs/snippets/documents/is-read-only.swift
import JsBaoClient

// Check whether a document is read-only for the current user (reader
// permission). Synchronous local read.
func isReadOnly(client: JsBaoClient, documentId: String) {
  // #region example
  let readOnly = client.documents.isReadOnly(documentId: documentId)
  // #endregion example
  _ = readOnly
}

File: dev-docs/snippets/documents/is-read-only.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Check whether a document is read-only for the current user.
export function isReadOnly(client: JsBaoClient, documentId: string) {
  // #region example
  const readOnly = client.documents.isReadOnly(documentId);
  // #endregion example
  return readOnly;
}
